nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -192,11 +222,28 @@ public boolean completeUpload(String batchId, String fileIndex, BatchFileInfo fi
         String digest = newMetadata.getContentMD5();
         String blobProviderId = transientStoreName; // TODO decouple this
         Blob blob = new BinaryBlob(new LazyBinary(blobKey, blobProviderId, null), blobKey, filename, mimeType, encoding,
-                digest, length);
+                                   digest, length
+        );
         Batch batch = getBatch(batchId);
         batch.addFile(fileIndex, blob, filename, mimeType);
 
         return true;
     }
 
+    protected String concatPath(String ... paths) {
+        StringBuilder stringBuilder = new StringBuilder();
+        for (String path : paths) {
+            stringBuilder.append(path);
+            if (!path.endsWith("/")) {
+                stringBuilder.append("/");
+            }
+        }
+
+        // Remove last "/"
+        return stringBuilder.substring(0, stringBuilder.length());;
+    }
+
+    private String replaceStringWithPlaceholders(String str, Pattern pattern, String replacement) {
+        return pattern.matcher(str).replaceAll(replacement);
+    }
 }
```

Comment:
Your whole class is badly formatted. at least 17 lines were formatted when I applied Nuxeo formatter. Please format !